shared/bee-game.js

```diff
@@ -112,10 +112,68 @@ function updateBoids(boids, { players }) {
   const teams = teamIds.map(teamId => ({
     boids: boidsAndPlayers.filter(boid => boid.teamId === teamId),
     allies: players.filter(player => player.teamId === teamId),
-  }))
+  }));
+
+  const { winner, loser } = duel(teams);
+  takeOverNearestBoid(winner, loser);
+
   return boids.map((boid) => updateBoid(boid, teams[boid.teamId]));
 }
 
+function duel(teams) {
+  if ( teamSpread(teams[0]) > teamSpread(teams[1]) ) {
+    return {
+      winner: teams[1],
+      loser: teams[0]
+    };
+  } else {
+    return {
+      winner: teams[0],
+      loser: teams[1]
+    };
+  }
+}
+
+function teamSpread(team) {
+  const xCoords = team.boids.map(b => b.position.x);
+  const yCoords = team.boids.map(b => b.position.y);
+  const boundingBox = {
+    width: Math.max(xCoords) - Math.min(xCoords),
+    height: Math.max(yCoords) - Math.min(yCoords)
+  };
+  boundingBox.surface = boundingBox.width * boundingBox.height;
+  return boundingBox.surface;
+}
+
+function teamCenter(team) {
+  let xCoords, yCoords;
+  if ( team.allies.length > 0 ) {
```

Comment:
What is the disadvantage of just checking boids (instead of allies and optionally boids).

Reply:
I want the "player bee" to have the actual impact, because that's what you control. But locally, with just one player, there is no "player bee" for the second team, so I take the boids' position in that case. So it's more a workaround.

Also, I'm not sure this is going to work nicely with multiple players in one team. But we'll see.